# BUSCO_phylogenomics (custom)

**Attantion:**
As BUSCO has pubulished the new protocol for building phylogenomic tress, no need for using the scripts in this repository. 
Please refer to the new protocol: <https://gitlab.com/ezlab/busco_protocol/>.
Detials can be found in the paper: <https://doi.org/10.1002/cpz1.323>, see Support Protocol 3: Building phylogenomic trees.

These scipts are the additional scripts related to Benchmarking Universal Single-Copy Orthologs (BUSCO).
Only need for someone has used the old protocool for phylogenomic analysis and just find the old scripts dont work with the new BUSCO output files (still recommend to use the new protocol for phylogenomic analysis from the beginningðŸ™ƒ).

For the original repository itself: <https://gitlab.com/ezlab/busco_usecases>

As the BUSCO has been updated to Version 5, the original scripts dont work with the new ouput files of the BUSCO run. So I modify the scripts for working with BUSCO v5.

**The citation for the BUSCO and the original scripts**

*BUSCO applications from quality assessments to gene prediction and phylogenomics.*
Robert M. Waterhouse, Mathieu Seppey, Felipe A. SimÃ£o, Mose Manni, Panagiotis Ioannidis, Guennadi Klioutchnikov, Evgenia V. Kriventseva, and Evgeny M. Zdobnov
*Mol Biol Evol*, published online Dec 6, 2017
doi: 10.1093/molbev/msx319

*BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs.*
Felipe A. SimÃ£o, Robert M. Waterhouse, Panagiotis Ioannidis, Evgenia V. Kriventseva, and Evgeny M. Zdobnov
*Bioinformatics*, published online June 9, 2015
doi: 10.1093/bioinformatics/btv351

Copyright (c) 2016-2018, Evgeny Zdobnov (<ez@ezlab.org>)
Licensed under the MIT license.

**NOTE:**
If you are not familiar with python and can not edit the code for your own work, you may choose other repository like <https://github.com/jamiemcg/BUSCO_phylogenomics> for construct species phylogenies using BUSCO proteins.
