These scipts are the additional scripts related to Benchmarking Universal Single-Copy Orthologs (BUSCO) 

For the original repository itself: https://gitlab.com/ezlab/busco_usecases

As the BUSCO has been updated to Version 5, the original scripts dont work with the new ouput files of the BUSCO run. So I modify the scripts for working with BUSCO v5.

**The citation for the BUSCO and the original scripts**

*BUSCO applications from quality assessments to gene prediction and phylogenomics.*
Robert M. Waterhouse, Mathieu Seppey, Felipe A. Simão, Mose Manni, Panagiotis Ioannidis, Guennadi Klioutchnikov, Evgenia V. Kriventseva, and Evgeny M. Zdobnov
*Mol Biol Evol*, published online Dec 6, 2017 
doi: 10.1093/molbev/msx319 

*BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs.*
Felipe A. Simão, Robert M. Waterhouse, Panagiotis Ioannidis, Evgenia V. Kriventseva, and Evgeny M. Zdobnov
*Bioinformatics*, published online June 9, 2015 
doi: 10.1093/bioinformatics/btv351

Copyright (c) 2016-2018, Evgeny Zdobnov (ez@ezlab.org)
Licensed under the MIT license.

**NOTE:**
If you are not familiar with python and can not edit the code for your own work, you may choose other repository like https://github.com/jamiemcg/BUSCO_phylogenomics for construct species phylogenies using BUSCO proteins.